test/integration/controlplane/transformation/secrets_transformation_test.go

```diff
@@ -99,6 +120,452 @@ func TestSecretsShouldBeTransformed(t *testing.T) {
 	}
 }
 
+type verifier interface {
+	verify(t *testing.T, err error)
+}
+
+type allowUnsafeMalformedObjectDeletionTestRunner struct {
+	// whether the feature is enabled
+	corruptObjDeleteEnabled bool
+	// whether encryption broke after the change
+	encryptionBrokenFn func(t *testing.T, got apierrors.APIStatus) bool
+	// what we expect for GET on the corrupt object, after encryption has
+	// broken but before the deletion
+	corruptObjGet verifier
+	// what we expect for DELETE on the corrupt object, after encryption has
+	// broken, but without setting the option to ignore store read error
+	corrupObjDeletNoIgnore verifier
+	// what we expect for DELETE on the corrupt object, after encryption has
+	// broken, with the option to ignore store read error enabled
+	// the user does not have the privilege to do unsafe delete
+	corrupObjDeleteWithoutPrivilege verifier
+	// what we expect for DELETE on the corrupt object, after encryption has
+	// broken, with the option to ignore store read error enabled
+	// this time, the user has the permission to do unsafe delete
+	corrupObjDeleteWithPrivilege verifier
+	// what we expect for GET on the corrupt object (post deletion)
+	corrupObjFinalGet verifier
+}
+
+func TestAllowUnsafeMalformedObjectDeletionFeature(t *testing.T) {
+	tests := []struct {
+		runner allowUnsafeMalformedObjectDeletionTestRunner
+	}{
+		{
+			runner: allowUnsafeMalformedObjectDeletionTestRunner{
+				corruptObjDeleteEnabled: true,
+				encryptionBrokenFn: func(t *testing.T, got apierrors.APIStatus) bool {
+					return got.Status().Reason == metav1.StatusReasonInternalError &&
+						strings.Contains(got.Status().Message, "StorageError: corrupt object")
+				},
+				corruptObjGet:          wantAPIStatusError{reason: metav1.StatusReasonInternalError},
```

Comment:
I think we should add a test that verifies that failed LIST return a list of corrupted objects. Maybe in a separate test? Or that already covered by unit tests?

Reply:
yes, I have a separate integration test for that: https://github.com/kubernetes/kubernetes/pull/127513/commits/c5d8d5ae850512ae340a25c32d611057da4d531d, please take a look :)